libfreerdp/codec/h264_mf.c

```diff
@@ -494,12 +494,15 @@ static BOOL mf_init(H264_CONTEXT* h264)
 	if (!sys)
 		goto error;
 
-	h264->pSystemData = (void*) sys;
 	/* http://decklink-sdk-delphi.googlecode.com/svn/trunk/Blackmagic%20DeckLink%20SDK%209.7/Win/Samples/Streaming/StreamingPreview/DecoderMF.cpp */
 	sys->mfplat = LoadLibraryA("mfplat.dll");
-
 	if (!sys->mfplat)
+	{
```

Comment:
What was this change necessary for?
`sys` was cleaned up in `mf_uninit` below.

Reply:
before, if the block was allocated mf_uninit blindly tries to access some fields that don't end up getting initialized, MFShutdown for one

Reply:
The free() here is not the way to go imho.
The problem is the current mf_uninit implementation:
- First it should not call MFShutdown without checking if the pointer is valid
- Secondly MFShutdown should not be called after mfplat.dll was freed with FreeLibrary

With these changes a goto error; should be sufficient here ...

Reply:
@mfleisz ok but there are more logic problems in the init/uninit than just those. I'll make a another pass and clean it up

Reply:
@perkerk thanks!